### PR TITLE
Add servicemonitor

### PR DESCRIPTION
**Pull request by QuentinBisson**
Servicemonitor will need to be pushed upstream but that's fine for now

### CODE REVIEW

File: helm/grafana-agent/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    {{- if .Values.serviceMonitor.interval }}
+    interval: {{ .Values.serviceMonitor.interval }}
+    {{- end }}
+{{- if .Values.serviceMonitor.metricRelabelings }}
+    metricRelabelings:
+{{ tpl (toYaml .Values.serviceMonitor.metricRelabelings | indent 6) . }}
+{{- end }}
+{{- if .Values.serviceMonitor.relabelings }}
+    relabelings:
+{{ toYaml .Values.serviceMonitor.relabelings | indent 6 }}
```
Comment (marieroque) [inline helm/grafana-agent/templates/servicemonitor.yaml]: Why do you use `tpl` with `metricRelabelings` and not with `relabelings` ?

File: helm/grafana-agent/templates/_helpers.tpl [hunk 2]
```diff
@@ -1,38 +1,27 @@
...
 giantswarm.io/service-type: {{ .Values.serviceType }}
-helm.sh/chart: {{ include "chart" . | quote }}
-{{- end -}}
+{{- end }}
 
+{{/*
+Selector labels
+*/}}
+{{- define "grafana-agent.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "grafana-agent.name" . }}
```
Comment (marieroque) [inline helm/grafana-agent/templates/_helpers.tpl:25]: Where is this value coming from ?
here https://github.com/grafana/helm-charts/tree/main/charts/grafana-agent ?
I see nothing in the dependency !?

Comment (QuentinBisson) [reply helm/grafana-agent/templates/_helpers.tpl:25]: It's coming from https://github.com/grafana/agent/blob/main/operations/helm/charts/grafana-agent/templates/_helpers.tpl

File: helm/grafana-agent/templates/servicemonitor.yaml (hunk 1 above)
Comment (QuentinBisson) [reply helm/grafana-agent/templates/servicemonitor.yaml]: I copied it from here https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus-operator/servicemonitor.yaml

Comment (QuentinBisson) [reply helm/grafana-agent/templates/servicemonitor.yaml]: Made it a tpl also

**Comment by QuentinBisson** [comment]
Tests are fixed :)

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**